test/jdk/java/awt/print/PrinterJob/PrintImage.java

```diff
@@ -21,276 +21,144 @@
+import java.awt.Font;
+import java.awt.Frame;
+import java.awt.Graphics;
+import java.awt.Menu;
+import java.awt.MenuBar;
+import java.awt.MenuItem;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.print.PageFormat;
+import java.awt.print.Printable;
+import java.awt.print.PrinterException;
+import java.awt.print.PrinterJob;
+
+/*
+ * @test
  * @bug 4298489
  * @summary Confirm that output is same as screen.
  * @key printer
+ * @requires os.family=="windows"
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
  * @run main/manual PrintImage
  */
-import java.awt.*;
-import java.awt.print.*;
-import java.awt.event.*;
-
 public class PrintImage extends Frame implements ActionListener {
-
-        private PrintImageCanvas                printImageCanvas;
-
-        private MenuItem        print1Menu = new MenuItem("PrintTest1");
-        private MenuItem        print2Menu = new MenuItem("PrintTest2");
-        private MenuItem        exitMenu = new MenuItem("Exit");
-
-        public static void main(String[] argv) {
-        String[] instructions =
-           { "You must have a printer available to perform this test,",
-             "prefererably Canon LaserShot A309GII.",
-             "Printing must be done in Win 98 Japanese 2nd Edition.",
-             "",
-             "Passing test : Output of text image for PrintTest1 and PrintTest2 should be same as that on the screen.",
-           };
-
-        Sysout.createDialog( );
-         Sysout.printInstructions( instructions );
-
-                new PrintImage();
-        }
-
-        public PrintImage() {
-                super("PrintImage");
-                initPrintImage();
-        }
-
-        public void initPrintImage() {
-
-                printImageCanvas = new PrintImageCanvas(this);
-
-                initMenu();
-
-                addWindowListener(new WindowAdapter() {
-                        public void windowClosing(WindowEvent ev) {
-                                dispose();
-                        }
-                        public void windowClosed(WindowEvent ev) {
-                                System.exit(0);
-                        }
-                });
-
-                setLayout(new BorderLayout());
-                add(printImageCanvas, BorderLayout.CENTER);
-                pack();
-
-                setSize(500,500);
-                setVisible(true);
+    private final PrintImageCanvas printImageCanvas = new PrintImageCanvas();
+    private final MenuItem print1Menu = new MenuItem("PrintTest1");
+    private final MenuItem print2Menu = new MenuItem("PrintTest2");
+    private static final String INSTRUCTIONS =
+            "Printing must be done in Japanese 2nd Edition.\n" +
+            "Preferably Canon LaserShot A309GII.\n" +
+            "\n" +
+            "Passing test : Output of text image for PrintTest1 and PrintTest2 should be\n" +
+            "same as that on the screen.";
```

Comment:
Expand the instructions:
```suggestion
            "Select PrintTest1 in the File menu.\n" +
            "Print Dialog will appear.\n" +
            "Click OK to start the first print job.\n" +
            "\n" +
            "Select PrintTest2 in the File menu.\n" +
            "Page Setup Dialog will appear.\n" +
            "Click OK.\n" +
            "Print Dialog will appear.\n" +
            "Click OK to start the second print job.\n" +
            "\n" +
            "The text in the printouts for PrintTest1 and PrintTest2 should be\n" +
            "same as that on the screen.\n" +
            "Press Pass if they are, otherwise press Fail.";
```